### PR TITLE
Update ElementRef example in RC/Blazor JS interop

### LINKED ISSUE #11607: How did you get this to compile?
this is a great idea but going from static to instanced variables seems like a stretch

namespace MyLib
{
    public static class MyLibElementRefExtensions
    {
        private readonly IJSRuntime _jsRuntime;

        public MyJsInterop(IJSRuntime jsRuntime)
        {
            _jsRuntime = jsRuntime;
        }

        public static Task Focus(this ElementRef elementRef)
        {
            return _jsRuntime.InvokeAsync&lt;object&gt;(
                "myLib.focusElement", elementRef);
        }
    }
}

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 93f4eeb7-4bf9-2631-3814-52bd4608445b
* Version Independent ID: fcde7f10-d897-1276-a3b7-4c5d19d14e4c
* Content: [Razor Components JavaScript interop](https://docs.microsoft.com/en-us/aspnet/core/razor-components/javascript-interop?view=aspnetcore-3.0#feedback)
* Content Source: [aspnetcore/razor-components/javascript-interop.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/razor-components/javascript-interop.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #10717 
Fixes #11607

@pfeurean @dharmaturtle *Now, we're cook'in with gas!* :smile: What do you think of the updates here? In addition to fixing it, I align it better with the sample app. In a future common sample update, I'll (probably) roll this into the sample app. For now tho, let's just get this update merged asap.

### CODE REVIEW

**Comment by dharmaturtle** [comment]
Addresses #11607 very nicely, thanks!

Just wondering, the `public async void SetFocus()` makes my neck hairs rise. Is this a safe spot to not return `Task`? I know in some places (like event handlers) that's accepted.

**Comment by guardrex** [comment]
Yes ... AFAIK.

I'm sure engineering will issue a proper 💀 **_death threat_** :skull: to me to change it if needed. :smile: lol

Sorry for the delay on it. The conference kept us all fairly busy last week, including the engineering cats. Thanks again for opening the issue and letting me know about the problem.

**Comment by guardrex** [comment]
@danroth27 When you get to this one, you might wish to look at the whole topic. There were several other Preview 3 updates on https://github.com/aspnet/Docs/pull/11314 that didn't get fully reviewed due to the conference.

File: aspnetcore/razor-components/javascript-interop.md [hunk 1]
```diff
@@ -210,57 +210,88 @@ As far as .NET code is concerned, an `ElementRef` is an opaque handle. The *only
...
 
 <input ref="username" />
-<button onclick="@SetFocus">Set focus</button>
+<button onclick="@SetFocus">Set focus on username</button>
 
 @functions {
     ElementRef username;
 
-    void SetFocus()
+    public async void SetFocus()
```
Comment (danroth27) [inline aspnetcore/razor-components/javascript-interop.md]: Return Task

File: aspnetcore/razor-components/javascript-interop.md [hunk 2]
```diff
@@ -210,57 +210,88 @@ As far as .NET code is concerned, an `ElementRef` is an opaque handle. The *only
...
         private readonly IJSRuntime _jsRuntime;
 
-        public MyJsInterop(IJSRuntime jsRuntime)
+        public ExampleJsInterop(IJSRuntime jsRuntime)
         {
             _jsRuntime = jsRuntime;
         }
 
-        public static Task Focus(this ElementRef elementRef)
+        public Task Focus(ElementRef elementRef)
```
Comment (danroth27) [inline aspnetcore/razor-components/javascript-interop.md]: What's the reasoning behind making these all instance methods?

**Comment by guardrex** [comment]
@danroth27 

> What's the reasoning behind making these all instance methods?

Probably just held over from the first version of the topic. I refactored the `Display` and `Prompt` methods out and dropped the `ElementRef` example that used an instance method.

It's ready for another :eyes: look.

**Pull request merged**